object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionVisitor.java

```diff
@@ -49,6 +63,23 @@ public Void visitAnnotation(final AnnotationTree node, final Void p) {
   @Override
   public Void visitMethodInvocation(MethodInvocationTree node, Void p) {
 
+    if (!isAssignedToLocal(this.getCurrentPath()) && !atypeFactory.returnsThis(node)) {
```

Comment:
We can leave this logic for now, but it will likely be much simpler or even gone in the future once we can track types of particular expressions in the dataflow analysis